doc/appendices/command-line/traffic_ctl.en.rst

```diff
@@ -258,27 +258,54 @@ traffic_ctl host
     Get the current status of the hosts used in parent.config as a next hop in a multi-tiered cache hierarchy.  The value 0 or 1 is returned indicating that the host is marked as down '0' or marked as up '1'.  If a host is marked as down, it will not be used as the next hop parent, another host marked as up will be chosen.
 
 .. program:: traffic_ctl host
-.. option:: down --time seconds --reason 'manual|active|local' HOSTNAME [HOSTNAME ...]
+.. option:: down --time seconds --reason 'all|active|local|manual|self_detect' HOSTNAME [HOSTNAME ...]
 
     Marks the listed hosts as down so that they will not be chosen as a next hop parent.
     If the --time option is included, the host is marked down for the specified number of
     seconds after which the host will automatically be marked up.  0 seconds marks the host
     down indefinitely until marked up manually and is the default. A reason tag may be used
-    when marking a host down.  Valid values are 'manual', 'active', or 'local', 'manual' is
-    used as the default.  The tags are used to indicate wehter the host was marked down
-    manually or by an 'active' or 'local' health check.  There are three reason tag
-    metrics for each host that may be viewed to see the reason a host was marked down.
+    when marking a host down.  Valid values are 'manual', 'active', 'local', 'self_detect',
+    or 'all', 'manual' is used as the default.  The tags are used to indicate wether the host 
+    was marked down manually or by an 'active' or 'local' health check.  'self_detect' indicates
+    that a parent entry in parent.config was marked down because the entry refers to the
+    local host so, it is automatically marked down to prevent requests from looping. A host is
+    not marked up until all reason codes are cleared by marking up the host for the specified
+    reason code. The 'all' reason should be used very carefully as in the case of marking up,
+    it would clear the 'self_detect' reason and could result in trasaction failures due to
+    looping.  
+    
+    A stat is created for each host, with a the host fqdn and is prefixed with the string 
+    `proxy.process.host_status` with a string value.  The string value is a 
+    serialized representation of the Host status struct showing all current data ie, reasons, 
+    marked down times, and down time for each host.  The stats may be viewed using the 
+    `traffic_ctl metric` command or through the `stats_over_http` endpoint.
 
 .. program:: traffic_ctl host
-.. option:: up --reason 'manual|active|local' HOSTNAME [HOSTNAME ...]
+.. option:: up --reason 'all|active|local|manual|self_detect' HOSTNAME [HOSTNAME ...]
```

Comment:
what is the reason to have self_detect as an option? shouldn't self_detect be identified by ATS automatically?

Reply:
It doesn't need to be an option for traffic_ctl but there could be a use case where you might want to clear it.  I could remove it along with the 'all' option to traffic_ctl.